integration/mediation-tests/tests-platform/tests-rabbitmq/src/test/java/org/wso2/carbon/esb/rabbitmq/inbound/RabbitMQInboundTestCase.java

```diff
@@ -0,0 +1,97 @@
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.esb.rabbitmq.inbound;
+
+import org.testng.Assert;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+import org.wso2.carbon.esb.rabbitmq.utils.RabbitMQServerInstance;
+import org.wso2.carbon.esb.rabbitmq.utils.RabbitMQTestUtils;
+import org.wso2.carbon.integration.common.admin.client.LogViewerClient;
+import org.wso2.esb.integration.common.utils.ESBIntegrationTest;
+import org.wso2.esb.integration.common.utils.clients.rabbitmqclient.RabbitMQProducerClient;
+import org.wso2.carbon.logging.view.stub.types.carbon.LogEvent;
+
+import java.io.File;
+
+/**
+ * Includes a test case which deploys a simple rabbitmq inbound endpoint and tests for message consumption through
+ * that.
+ */
+public class RabbitMQInboundTestCase extends ESBIntegrationTest {
+
+    private LogViewerClient logViewer;
+    private RabbitMQProducerClient sender;
+
+    @BeforeClass(alwaysRun = true)
+    public void init() throws Exception {
+        super.init();
+        sender = RabbitMQServerInstance.createProducerWithDeclaration("exchange", "simple_inbound_endpoint_test");
+        //The inbound endpoint cannot be pre-deployed since the queue declaration(which is done in 'initRabbitMQBroker')
+        // must happen before deployment and also because it will cause unnecessary message polling.
+        loadESBConfigurationFromClasspath(File.separator + "artifacts" + File.separator + "ESB" + File.separator
+                                          + "inbound" + File.separator + "rabbitmq_inbound_endpoint.xml");
+        logViewer = new LogViewerClient(contextUrls.getBackEndUrl(), getSessionCookie());
+    }
+
+    /**
+     * Publishes 200 messages to the queue 'simple_inbound_endpoint_test' and asserts for the log
+     * 'received by inbound endpoint = true' which is logged once a message is picked up by the inbound endpoint.
+     *
+     * @throws Exception if an error occurs while accessing the logViewer client or while publishing messages.
+     */
+    @Test(groups = {"wso2.esb"}, description = "Test ESB as a RabbitMQ inbound endpoint ")
+    public void testRabbitMQInboundEndpoint() throws Exception {
+
+        logViewer.clearLogs();
+        int messageCount = 200;
+
+        String message = "<ser:placeOrder xmlns:ser=\"http://services.samples\">\n" + "<ser:order>\n"
+                         + "<ser:price>100</ser:price>\n" + "<ser:quantity>2000</ser:quantity>\n"
+                         + "<ser:symbol>RMQ</ser:symbol>\n" + "</ser:order>\n" + "</ser:placeOrder>";
+        for (int i = 0; i < messageCount; i++) {
+            sender.sendMessage(message, "text/plain");
+        }
+
+        RabbitMQTestUtils.waitForLogToGetUpdated();
```

Comment:
Are we waiting here log get updated in rabbitMQ or ESB? 
If we wait for ESB log update, I see only thread sleep waitForLogToGetUpdated() function is it possible to add function to LogViewerClient to wait for log update actually reading the log (with timeout)?

Reply:
LogViewerclient is coming from org.wso2.carbon.integration.common.admin.client and it is simply calling a webservice to get the logs. It might sometimes not make send to have a thread sleep at that level. WDYT?

Reply:
Ah, then I think this is fine, I thought LogViewerclient capture log from file system or stdout. 
BTW, the polling time inbound endpoint is less than 5000ms? because if timming not enough possibility of test failure depending on the test environment

Reply:
There is no polling interval in rabbitMQ inbound, it receives messages whenever avaliable

Reply:
Ah got it ... 
just checked doc, since RabbitMQ Inbound Protocol is event-based inbound, only polls once to establish connection 
https://docs.wso2.com/display/EI611/Event-Based+Inbound+Endpoints

Thanks,